@coderabbitai CI fails with:
```
Error: src/esp_loader.ts(2424,5): error TS2322: Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
  Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer'.
    Types of property '[Symbol.toStringTag]' are incompatible.
      Type '"SharedArrayBuffer"' is not assignable to type '"ArrayBuffer"'.
```
your review did not found

--- a/src/esp_loader.ts
+++ b/src/esp_loader.ts
@@ -2421,7 +2421,7 @@ export class ESPLoader extends EventTarget {
     }
 
     const paddedData = padTo(new Uint8Array(binaryData), 4);
-    binaryData = paddedData.buffer;
+    binaryData = paddedData.buffer as ArrayBuffer;
 
     const uncompressedFilesize = binaryData.byteLength;
     let compressedFilesize = 0;

--- a/src/util.ts
+++ b/src/util.ts
@@ -58,7 +58,11 @@ export const formatMacAddr = (macAddr: number[]): string => {
  * @name padTo
  * Pad data to the next alignment boundary with the given fill byte (default 0xFF)
  */
-export function padTo(data: Uint8Array, alignment: number, padCharacter = 0xff): Uint8Array {
+export function padTo(
+  data: Uint8Array,
+  alignment: number,
+  padCharacter = 0xff,
+): Uint8Array {
   const padMod = data.length % alignment;
   if (padMod !== 0) {
     const padding = new Uint8Array(alignment - padMod).fill(padCharacter);

